### ISSUE TITLE
Parallelize Github Actions steps to speed up the build

### ISSUE THREAD

**Issue by kevinrolsen** [gitcoinco/web#10048]
Let's improve the visibility into our build and speed things up, comment from the original story:

> "I think it would be great if we could split this up into three or four jobs. Everything would have to wait on the build job but we could create a separate job for the cypress and python tests that could run in parallel which could speed things up a little. It could also give a little more transparency into a test failure. https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions#the-components-of-github-actions"

**Comment by nutrina** [comment]
FYI: I have been working on the Cypress tests on this branch: https://github.com/gitcoinco/web/pull/9979 (this is based on the initial work of @ksolo )

I have created a metamask stub, which allows us to run the tests in headless mode.
I am facing an issue however: the cypress tests hang after a while and do not finish.

I think we should first fix this before parallelising tests.
